red/Animal/lib/RaceHorse.pm

```diff
@@ -0,0 +1,183 @@
+package RaceHorse;
+
+use 5.006;
+use strict;
+use warnings;
+use Carp;
+use parent 'Horse';
+
+=head1 NAME
+
+RaceHorse - an accurate depiction of a race Horse.
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+dbmopen( our %HORSES, 'racehorses', 0666 )
+  or croak 'Failed to open racehorse database with read/write permissions';
+
+=head1 SYNOPSIS
+
+Maintains the records of a race Horse.
+
+    use RaceHorse;
+
+    my $horse = RaceHorse->new( name => 'Existential Crisis Pony' );
+    $horse->won;
+
+=head1 SUBROUTINES/METHODS
+
+=cut
+
+sub valid_properties { qw( wins places shows losses ) }
```

Comment:
Nah, need this due to me showing that you can more heavily protect instance state in the previous version. I was tempted to rip it out and start again, but figured it'd be more interesting to try and get everything working.